src/algorithm/mod.rs

```diff
@@ -50,7 +50,7 @@ impl Algorithm {
 
     /// Set the initial Node to a new state using the provided Goban.
     pub fn update_initial_state(&mut self, initial_state: Goban, last_move: BitBoard, player_captures: u8, opponent_captures:u8) {
-        let new_initial_node = Node::new(initial_state, 0, last_move, false, player_captures, opponent_captures);
+        let new_initial_node = Node::new(initial_state, crate::DEPTH as usize, last_move, false, player_captures, opponent_captures);
```

Comment:
If this function is called a lot of times it might not be worth casting this every time, we could change DEPTH's type to an usize

Reply:
You're right, nice catch, the method itself is not called a lot, thus I don't think this cast is an issue for the perf, however, it's not clean code and we should avoid cast as much as possible.

I'll fix it! 👍